controller/server_controller.py

```diff
@@ -125,31 +123,45 @@ def cull_idle_jupyter_servers(body, name, namespace, logger, **kwargs):
     js_server_status = get_js_server_status(body)
     custom_resource_api = get_api(config.api_version, config.custom_resource_name)
     idle_seconds = int(body["status"].get("idleSeconds", 0))
+    now = pytz.UTC.localize(datetime.utcnow())
+    last_activity = now if js_server_status is None else js_server_status.get("last_activity", now)
+    try:
+        jupyter_server_started = datetime.fromisoformat(
```

Comment:
Do you know if the last activity is always reported in UTC? Is this a setting we can explicitly enforce at the Jupyter server level to avoid some non-nice surprises?

Reply:
@ableuler I am not sure. I will check. Only UTC (i.e. Zulu time) gets a `Z` at the end in timestamps. The problem is that the python implementation of `fromisoformat` does not accept this - it always wants a specific offset. Other time zones will have a specific offset (i.e. `+02:00`) and in this case they can be parsed.

Reply:
I think that jupyterlab uses whatever time zone is set in the image that runs the server. I looked and I cannot find any definite documentation on this.

I think that for now I can make the code a bit more conservative by doing something like:
```if time.endswith("Z"): time = time[:-1] + "+00:00"```

Rather than using just `.replace`.